sklearn/utils/__init__.py

```diff
@@ -506,3 +508,146 @@ def indices_to_mask(indices, mask_length):
     mask[indices] = True
 
     return mask
+
+
+def flexible_vstack(it, final_len=None):
```

Comment:
This is not strictly necessary to the feature, and might be too magical. It merely helps users making use of `pairwise_distances_chunked`.

Reply:
I've had a go at removing `flexible_vstack`. In some cases it needs to be replaced by a couple of lines of code, but in radius_neighbors it gets pretty hairy, because it's dealing with arrays of arrays which do not `vstack` nicely.